src/test/java/com/example/dog/dao/JdbcDogDAOTest.java

```diff
@@ -0,0 +1,62 @@
+package com.example.dog.dao;
+
+import com.example.dog.model.Dog;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+@ContextConfiguration(locations={"classpath:/controllers-context.xml"})
+public class JdbcDogDAOTest extends AbstractTestNGSpringContextTests {
+
+    @Autowired
+    private JdbcDogDAO jdbcDogDao;
+
+    @Test
+    public void possibleToSaveNameWithMaxLength() {
+        Dog dog = new Dog(
+                "QWERTYUIOPASDFGHJKLZXCVBNMQWERQWERTYUIOPASDFGHJKLZXCVBNMQWERQWERTYUIOPASDFGHJKLZXCVBNMQWERTYUIOPASDFG",
+                24, 8,
+                ZonedDateTime.of(LocalDateTime.of(2020, Month.OCTOBER, 26, 4, 59),
+                        ZoneId.of("Europe/Moscow")));
+        Assert.assertEquals(jdbcDogDao.add(dog), dog);
+    }
+
+    @Test
+    public void possibleToSaveDogWithSQLInjectedName() {
+        Dog dog = new Dog("' blah", 24, 8,
+                ZonedDateTime.of(LocalDateTime.of(2020, Month.OCTOBER, 26, 4, 59),
+                        ZoneId.of("Europe/Moscow")));
+        Assert.assertEquals(jdbcDogDao.add(dog), dog);
```

Comment:
Let's add unitils as a dependency and use its `assertReflectionEquals()`. Since we haven't finished our version :)